org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/declaration/JvmBasedDeclarations.xtend

```diff
@@ -1320,6 +1320,8 @@ class JvmFieldDeclarationImpl extends JvmMemberDeclarationImpl<JvmField> impleme
 	}
 	
 	override remove() {
+		compilationUnit.jvmModelAssociator.removeAllAssociation(delegate.type)
```

Comment:
There might be other `removevAllAssociations(..)` missing in this class, no?

Reply:
yes probably on other kind of remove 
but since it make silently crash other remove (see the new failed test. after the first remove. the other ones aren't processed)

I'm not sure anymore that it should be done that way

I must admit I don't master xtend internal structures and I'm not efficient in finding a solution for cleaning these structures on all remove situations

Reply:
I'll give it a stab later this week.